test/Npgsql.Tests/SystemTransactionTests.cs

```diff
@@ -341,7 +489,7 @@ public void SchemaConnectionShouldntEnlist()
         }
 
         void AssertNoPreparedTransactions()
-            => Assert.That(GetNumberOfPreparedTransactions(), Is.EqualTo(0));
+            => Assert.That(GetNumberOfPreparedTransactions(), Is.EqualTo(0), "Prepared transactions found");
 
         int GetNumberOfPreparedTransactions()
```

Comment:
Well I guess I misunderstand it. I think this counter can go-up only when the prepare phase has been executed, but is not yet committed or rollback. So we cannot see that count in most circumstances. I do not think it is a good indicator for knowing if a distributed transaction is ongoing. Better check `Transaction.Current.TransactionInformation.DistributedIdentifier != Guid.Empty`.